src/event/getStream.ts

```diff
@@ -0,0 +1,27 @@
+import { Contract } from 'soroban-client';
+import getAdmin from '../utils/soroban/getAdmin';
+import getServer from '../utils/soroban/getServer';
+import buildGetStreamTransaction from '../utils/soroban/stream/buildGetStream';
+import Stream from '../models/Stream';
+import bigintToNumberStream from '../utils/soroban/stream/bigintToNumberStream';
+
+const getStreamAndSave = async (id: string) => {
+  const server = getServer();
+  const accountAdmin = await server.getAccount(getAdmin().publicKey());
+  const contract = new Contract(String(process.env.CONTRACT_ID));
+  const getStream = await buildGetStreamTransaction(accountAdmin, contract, id);
+
+  const existingStream = await Stream.findOne({ id_stream: id });
+  if (existingStream) {
+    return 'stream in db';
+  }
+  const streamDetails = bigintToNumberStream(getStream);
+  streamDetails.id_stream = Number(id);
+  streamDetails.status = 'ongoing';
```

Comment:
create a function called `calculateStreamStatus` that takes the stream and returns an `status`. And check `start_date` and `end_date` of an stream to see if it's `pending`, `ongoing`, or `expired`